app/src/main/java/ch/sdp/vibester/activity/DownloadActivity.kt

```diff
@@ -61,6 +61,23 @@ class DownloadActivity : AppCompatActivity() {
         registerReceiver(broadcast, IntentFilter(DownloadManager.ACTION_DOWNLOAD_COMPLETE))
     }
 
+    private fun downloadListener(songView: TextView) {
+        downloadComplete = false
+        songName = songView.text.toString()
+
+        if(checkExistingSong()) {
+            alert("Song already downloaded!", "Try a different song!", songView)
```

Comment:
Consider putting string message in string.xml

Reply:
Done